### PR TITLE
adds WithdrawablePeriphery base contract for token withdrawals

### PR DESCRIPTION
**Pull request by 0xDEnYO**
# Which Jira task belongs to this PR?
https://lifi.atlassian.net/browse/LF-10387

# Why did I implement it this way?

# Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: src/Helpers/WithdrawablePeriphery.sol [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    function withdrawToken(
+        address assetId,
+        address payable receiver,
+        uint256 amount
+    ) external onlyOwner {
+        if (LibAsset.isNativeAsset(assetId)) {
+            // solhint-disable-next-line avoid-low-level-calls
+            (bool success, ) = receiver.call{ value: amount }("");
+            if (!success) revert ExternalCallFailed();
+        } else {
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply src/Helpers/WithdrawablePeriphery.sol:30]: too late, the contract already got audited as-is. But you can suggest this change in the future again on other occassion

File: test/solidity/Helpers/WithdrawablePeriphery.t.sol [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+contract WithdrawablePeripheryTest is TestBase {
+    WithdrawablePeriphery internal withdrawable;
+    event TokensWithdrawn(
+        address assetId,
+        address payable receiver,
+        uint256 amount
+    );
+    error UnAuthorized();
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply test/solidity/Helpers/WithdrawablePeriphery.t.sol]: outdated

File: test/solidity/Helpers/WithdrawablePeriphery.t.sol [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        vm.expectRevert(UnAuthorized.selector);
+
+        withdrawable.withdrawToken(
+            ADDRESS_USDC,
+            payable(nonETHReceiver),
+            withdrawAmount
+        );
+    }
+}
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply test/solidity/Helpers/WithdrawablePeriphery.t.sol]: outdated

File: test/solidity/Helpers/WithdrawablePeriphery.t.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply test/solidity/Helpers/WithdrawablePeriphery.t.sol:23]: That's just the way this is across our whole codebase. Do not report this again.

**Review by ezynda3 — APPROVED** [review]

**Pull request merged**